src/Services/SocialUserLoginService.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core\Services;
+
+use Blumilk\Meetup\Core\Models\User;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Contracts\Hashing\Hasher;
+use Illuminate\Session\Store;
+
+class SocialUserLoginService
+{
+    public function __construct(
+        public AuthManager $authManager,
+        public Hasher $hasher,
+        public Store $session,
+    ) {}
+
+    public function registerOrLogin($socialUser, string $provider): void
+    {
+        $appUser = User::query()->firstOrCreate(
+            [
+                "email" => $socialUser["email"],
+            ],
+            [
+                "name" => $socialUser["name"],
+                "email" => $socialUser["email"],
```

Comment:
instead of array, u can use `Laravel\Socialite\Contracts\User;` and methods like below on the screenshot:

![obraz](https://user-images.githubusercontent.com/29115024/159501341-3527364d-719d-479f-8d9d-75a8804fc2cc.png)